Axom spack recipe can make bad host-config.

I was building Axom with spack with RAJA and without umpire. I ran into a problem because it did not include `CAMP_DIR` in the generated host-config used to build Axom.

```
  ./spack/bin/spack spec axom+raja~umpire %gcc
  ./spack/bin/spack install axom+raja~umpire %gcc
```

Leads to:
```
==> Installing axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a [56/56]
==> Ran patch() for axom
==> axom: Executing phase: 'initconfig'
==> axom: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/cmake-3.31.11-bdmg7b4lfhdchhribjuqkhgq2difse33/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a/lib;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a/lib64' '-DCMAKE_PREFIX_PATH:STRING=/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/blt-0.7.1-43jaityr56zmjvae7ginrfuzwvm43qy4;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/cmake-3.31.11-bdmg7b4lfhdchhribjuqkhgq2difse33;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-sapphirerapids/compiler-wrapper-1.0-f64j4mghi6qh7tk4oylmhxqfl5kvysc7;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/conduit-0.9.5-kp4e6mnl47cxf3orbxxdgyliixb6tniy;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/gmake-4.4.1-37nqagvbyeccmoame23omb7rmjz2nreg;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/lua-5.4.8-5knhe2cs63ci2wezf4tm43qhwgsixazk;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/raja-2025.12.2-qcb5m2gvxkhfxfsjh62r4nspvl3vtd7v;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/hdf5-1.8.23-w2rixfftwifenbvcjgnw6rwthupi65hf;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/parmetis-4.0.3-tlv74dqjf5fbl6pw2aghfepm3xowlecg;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/readline-8.3-4mz6thtrxt4er6h5hqs7kcxwg5wphhpf;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/camp-2025.12.0-r7ofhgh2owrhwyk5jj4zde3s5ofqcuno;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/metis-5.1.0-7aydxs4qvzhmtauhukscgwh2bfmo7be3;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/openmpi-5.0.10-d4zdb6k5cb73fijuyl6wuajdjf7zqh4m;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/numactl-2.0.19-wtypx7cgp4fllnuzqkbev36pm7xdgivy;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/prrte-4.1.0-s25ifesef7l7h2cek5zjeuy2atjcvgrn;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/pmix-6.1.0-5q6ydzemgxjekkni2i6f7bs72kxijrs2;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/hwloc-2.13.0-z3b7o5rkaetctsh7is7louygnnjk2gkc;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/libevent-2.1.12-dqjsp244xwp2eptrn6omqsnxutbtnrpb;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/libpciaccess-0.17-q7vnpnmg5rixc2r6qjerlekixigcs4on;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/libxml2-2.15.1-5xcsya7afmt6hygvjfyicgn3cuwcd67b;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/ncurses-6.6-dx7v4qphbx24zx7qp24wfxcijnru5edz;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/openssl-3.6.1-kcajji32ptcqulyz577lnqjapjgwuqjn;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/libiconv-1.18-uit44wvnm2eck27j7pn3zj52a6l2om6o;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/xz-5.8.2-pxzvn4ijao2ljga5ezd3mtyxln4hqqda;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/zlib-ng-2.3.3-dkf5mycmwwqihsvvmpebp6js6dj34r6v;/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/gcc-runtime-10.3.1-5mxniam3swmq5rr7jnwdrcvtjrsuiobx;/usr/tce' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-C' '/var/tmp/USER/spack-stage/spack-stage-axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a/spack-src/rzwhippet-toss_4_x86_64_ib-gcc@10.3.1.cmake' '-DBLT_SOURCE_DIR:PATH=/usr/WS2/USER/Axom/axom_spack/spack/opt/spack/linux-icelake/blt-0.7.1-43jaityr56zmjvae7ginrfuzwvm43qy4' '-DENABLE_TESTS=OFF' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DAXOM_ENABLE_EXAMPLES:BOOL=ON' '-DAXOM_ENABLE_TOOLS:BOOL=ON' '-DAXOM_ENABLE_TUTORIALS:BOOL=ON' '/tmp/USER/spack-stage/spack-stage-axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a/spack-src/src'

1 error found in build log:
     76    -- CMake's C implicit link directories: /collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc/x86_64-redhat-linux/10;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib
           64;/lib64;/usr/lib64;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib
     77    -- CMake's CXX implicit link directories: /collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc/x86_64-redhat-linux/10;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/l
           ib64;/lib64;/usr/lib64;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib
     78    -- CMake's Fortran implicit link directories: /collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc/x86_64-redhat-linux/10;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib/gcc;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3
           .1/lib64;/lib64;/usr/lib64;/collab/usr/global/tools/tce4/packages/gcc/gcc-10.3.1/lib
     79    -- CMake's C implicit link libraries: gcc;gcc_s;c;gcc;gcc_s
     80    -- CMake's CXX implicit link libraries: stdc++;m;gcc_s;gcc;c;gcc_s;gcc
     81    -- CMake's Fortran implicit link libraries: gfortran;m;gcc_s;gcc;quadmath;m;gcc_s;gcc;c;gcc_s;gcc
  >> 82    CMake Error at cmake/thirdparty/SetupAxomThirdParty.cmake:34 (message):
     83      CAMP_DIR is required if RAJA_DIR or UMPIRE_DIR is provided.
     84    Call Stack (most recent call first):
     85      cmake/CMakeBasics.cmake:20 (include)
     86      CMakeLists.txt:126 (include)
     87    
     88    

See build log for details:
  /var/tmp/USER/spack-stage/spack-stage-axom-0.12.0-qypbq6pxt36i6xe4sxnk64gwv5ctkf5a/spack-build-out.txt
```

--- a/scripts/spack/packages/axom/package.py
+++ b/scripts/spack/packages/axom/package.py
@@ -16,6 +16,8 @@ from spack_repo.builtin.build_systems.cached_cmake import (
 from spack_repo.builtin.build_systems.cuda import CudaPackage
 from spack_repo.builtin.build_systems.rocm import ROCmPackage
 
+from spack.package import *
+
 # Axom components we expose to Spack. Core is always built and is not listed here.
 _AXOM_COMPONENTS = (
     "bump",
@@ -33,9 +35,6 @@ _AXOM_COMPONENTS = (
     "slic",
     "spin",
 )
-
-
-from spack.package import *
 
 
 def get_spec_path(spec, package_name, path_replacements={}, use_bin=False, use_lib=False):
@@ -145,20 +144,21 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
         values=any_combination_of("all", *_AXOM_COMPONENTS).with_default("all"),
     )
 
+    variant("int64", default=True, description="Use 64bit integers for IndexType")
+
     # variants for package dependencies
     variant("adiak", default=False, when="@0.13:", description="Build with adiak")
-    variant("caliper", default=False, when="@0.13:", description="Build with caliper")
     variant("c2c", default=False, description="Build with c2c")
+    variant("caliper", default=False, when="@0.13:", description="Build with caliper")
     variant("conduit", default=True, description="Build with conduit")
     variant("hdf5", default=True, description="Build with hdf5")
     variant("lua", default=True, description="Build with Lua")
     variant("mfem", default=False, description="Build with mfem")
     variant("opencascade", default=False, description="Build with opencascade")
-    variant("raja", default=True, description="Build with raja")
     variant("scr", default=False, description="Build with SCR")
     variant("umpire", default=True, description="Build with umpire")
 
-    variant("int64", default=True, description="Use 64bit integers for IndexType")
+    variant("raja", default=True, description="Build with raja")
 
     varmsg = "Build development tools (such as Sphinx, Doxygen, etc...)"
     variant("devtools", default=False, description=varmsg)
@@ -176,10 +176,10 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("cmake@3.21:", type="build", when="+rocm")
 
     depends_on("blt", type="build")
-    depends_on("blt@0.5.1:0.5.3", type="build", when="@0.6.1:0.8")
-    depends_on("blt@0.6.2:", type="build", when="@0.9:")
-    depends_on("blt@0.7", type="build", when="@0.11:")
     depends_on("blt@0.7.1:", type="build", when="@0.12:")
+    depends_on("blt@0.7", type="build", when="@0.11:")
+    depends_on("blt@0.6.2", type="build", when="@0.9:0.10")
+    depends_on("blt@0.5.1:0.5.3", type="build", when="@0.6.1:0.8")
 
     depends_on("mpi", when="+mpi")
 
@@ -201,8 +201,10 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     with when("+umpire"):
         depends_on("umpire")
         depends_on("umpire@2025.12:", when="@0.13:")
-        depends_on("umpire@2025.09.0:", when="@0.10:")
-        depends_on("umpire@2024.02.0:", when="@0.9:")
+        depends_on("umpire@2025.09:", when="@0.12:")
+        depends_on("umpire@2025.03", when="@0.11")
+        depends_on("umpire@2024.07", when="@0.10")
+        depends_on("umpire@2024.02", when="@0.9")
         depends_on("umpire@2022.03.0:2023.06", when="@0.7.0:0.8")
         depends_on("umpire@6.0.0", when="@0.6.0")
         depends_on("umpire@5:5.0.1", when="@:0.5.0")
@@ -212,8 +214,10 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     with when("+raja"):
         depends_on("raja")
         depends_on("raja@2025.12.1:", when="@0.13:")
-        depends_on("raja@2025.09.0:", when="@0.10:")
-        depends_on("raja@2024.02.0:", when="@0.9:")
+        depends_on("raja@2025.09:", when="@0.12:")
+        depends_on("raja@2025.03", when="@0.11")
+        depends_on("raja@2024.07", when="@0.10")
+        depends_on("raja@2024.02", when="@0.9")
         depends_on("raja@2022.03.0:2023.06", when="@0.7.0:0.8")
         depends_on("raja@0.14.0", when="@0.6.0")
         depends_on("raja@:0.13.0", when="@:0.5.0")
@@ -226,7 +230,7 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("caliper", when="+caliper")
     with when("+profiling"):
         depends_on("adiak")
-        depends_on("caliper")
+        depends_on("caliper+adiak~papi")
 
         depends_on("caliper+cuda", when="+cuda")
         depends_on("caliper~cuda", when="~cuda")
@@ -254,7 +258,6 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
         depends_on(f"umpire {ext_cuda_dep}", when=f"+umpire {ext_cuda_dep}")
         depends_on(f"caliper {ext_cuda_dep}", when=f"+caliper {ext_cuda_dep}")
         depends_on(f"caliper {ext_cuda_dep}", when=f"+profiling {ext_cuda_dep}")
-
         depends_on(f"mfem {ext_cuda_dep}", when=f"+mfem {ext_cuda_dep}")
 
     for val in ROCmPackage.amdgpu_targets:
@@ -415,11 +418,14 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
         if spec.satisfies("%cce"):
             entries.append(cmake_cache_string("CMAKE_CXX_FLAGS_DEBUG", "-O1 -g"))
 
+            # Remove unusable -Mfreeform flag injected by spack
+            entries = [entry.replace("-Mfreeform", "") for entry in entries]
+
+        # Disable intrusive warning:
+        #   icpx: remark: note that use of '-g' without any optimization-level
+        #   option will turn off most compiler optimizations similar to use of
+        #   '-O0'; use '-Rno-debug-disables-optimization' to disable this remark
         if spec.satisfies("%oneapi"):
-            # Disable intrusive warning:
-            #   icpx: remark: note that use of '-g' without any optimization-level
-            #   option will turn off most compiler optimizations similar to use of
-            #   '-O0'; use '-Rno-debug-disables-optimization' to disable this remark
             entries.append(
                 cmake_cache_string("CMAKE_CXX_FLAGS_DEBUG", "-g -Rno-debug-disables-optimization")
             )
@@ -496,9 +502,8 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
 
             # Additional library path for cray compiler
             if spec.satisfies("%cce"):
-                hip_link_flags += "-L/opt/cray/pe/cce/{0}/cce/x86_64/lib -Wl,-rpath,/opt/cray/pe/cce/{0}/cce/x86_64/lib ".format(
-                    spec.compiler.version
-                )
+                lib_path = "/opt/cray/pe/cce/{0}/cce/x86_64/lib".format(spec.compiler.version)
+                hip_link_flags += "-L{0} -Wl,-rpath,{0}".format(lib_path)
 
             if spec.satisfies("+fortran"):
                 link_remove_list = []
@@ -698,7 +703,7 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
             else:
                 entries.append(f"# {dep.upper()} not built\n")
 
-        if spec.satisfies("+umpire") and spec.satisfies("^camp"):
+        if (spec.satisfies("+raja") or spec.satisfies("+umpire")) and spec.satisfies("^camp"):
             dep_dir = get_spec_path(spec, "camp", path_replacements)
             entries.append(cmake_cache_path("CAMP_DIR", dep_dir))
 
@@ -750,7 +755,6 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
             entries.append(cmake_cache_option("ENABLE_CLANGFORMAT", False))
 
         if spec.satisfies("+python") or spec.satisfies("+devtools"):
-            # Get path to python executable
             python_bin_dir = get_spec_path(spec, "python", path_replacements, use_bin=True)
             entries.append(cmake_cache_path("Python_EXECUTABLE", pjoin(python_bin_dir, "python3")))
 
@@ -837,7 +841,7 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
             print("Running Axom Unit Tests...")
             make("test")
 
-    @run_after("install")
+    @run_after("install", when="+examples")
     @on_package_attributes(run_tests=True)
     def test_install_using_cmake(self):
         """build example with cmake and run"""
@@ -853,7 +857,7 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
             example()
             make("clean")
 
-    @run_after("install")
+    @run_after("install", when="+examples")
     @on_package_attributes(run_tests=True)
     def test_install_using_make(self):
         """build example with make and run"""

--- a/scripts/spack/packages/axom/package.py
+++ b/scripts/spack/packages/axom/package.py
@@ -155,10 +155,9 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     variant("lua", default=True, description="Build with Lua")
     variant("mfem", default=False, description="Build with mfem")
     variant("opencascade", default=False, description="Build with opencascade")
+    variant("raja", default=True, description="Build with raja")
     variant("scr", default=False, description="Build with SCR")
     variant("umpire", default=True, description="Build with umpire")
-
-    variant("raja", default=True, description="Build with raja")
 
     varmsg = "Build development tools (such as Sphinx, Doxygen, etc...)"
     variant("devtools", default=False, description=varmsg)
@@ -230,7 +229,7 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("caliper", when="+caliper")
     with when("+profiling"):
         depends_on("adiak")
-        depends_on("caliper+adiak~papi")
+        depends_on("caliper+adiak")
 
         depends_on("caliper+cuda", when="+cuda")
         depends_on("caliper~cuda", when="~cuda")

